### PR TITLE
FOUNDATIONS: Modify `Time`

### LINKED ISSUE #225: FOUNDATIONS: Modify `Time`

### PR DESCRIPTION
**Pull request by MirzayevDeveloper**
closes #225

### CODE REVIEW

File: Tarteeb.Api/Services/Foundations/TimeSlots/TimeService.Validations.cs [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+            Validate(
+                (Rule: IsInvalid(time.Id), nameof(Time.Id)),
+                (Rule: IsInvalid(time.HoursWorked), nameof(Time.HoursWorked)),
+                (Rule: IsInvalid(time.Comment), nameof(Time.Comment)),
+                (Rule: IsInvalid(time.UserId), nameof(Time.UserId)),
+                (Rule: IsInvalid(time.TicketId), nameof(Time.TicketId)),
+                (Rule: IsInvalid(time.CreatedDate), nameof(Time.CreatedDate)),
+                (Rule: IsInvalid(time.UpdatedDate), nameof(Time.UpdatedDate)),
+                (Rule: IsInvalid(time.Ticket), nameof(Time.Ticket)),
+                (Rule: IsInvalid(time.User), nameof(Time.User)),
```
Comment (ElbekDeveloper) [inline Tarteeb.Api/Services/Foundations/TimeSlots/TimeService.Validations.cs]: Bularni tekshirish kerak emas

File: Tarteeb.Api.Tests.Unit/Services/Foundations/TimeSlots/TimeServiceTests.Exceptions.Modify.cs [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+                broker.GetCurrentDateTime()).Returns(randomDateTime);
+
+            // when 
+            ValueTask<Time> modifyTimeTask = this.timeService.ModifyTimeAsync(someTime);
+
+            TimeDependencyValidationException actualTimeDependencyValidationException =
+                await Assert.ThrowsAsync<TimeDependencyValidationException>(modifyTimeTask.AsTask);
+
+            // then 
+            actualTimeDependencyValidationException.Should().
```
Comment (Javlonbek-dev) [inline Tarteeb.Api.Tests.Unit/Services/Foundations/TimeSlots/TimeServiceTests.Exceptions.Modify.cs:149]: Nuqta BeEquivalentTo bilan birga pastga tushadi

File: Tarteeb.Api.Tests.Unit/Services/Foundations/TimeSlots/TimeServiceTests.Validation.Modify.cs [hunk 3]
```diff
@@ -0,0 +1,382 @@
...
+                broker.GetCurrentDateTime(), Times.Once);
+
+            this.loggingBrokerMock.Verify(broker =>
+                broker.LogError(It.Is(SameExceptionAs(
+                    expectedTimeValidationException))), Times.Once);
+
+            this.storageBrokerMock.Verify(broker =>
+                broker.SelectTimeByIdAsync(invalidTime.Id), Times.Once);
+
+            this.dateTimeBrokerMock.VerifyNoOtherCalls();
```
Comment (Javlonbek-dev) [inline Tarteeb.Api.Tests.Unit/Services/Foundations/TimeSlots/TimeServiceTests.Validation.Modify.cs:377]: storageBroker birinchi yoziladi

File: Tarteeb.Api.Tests.Unit/Services/Foundations/TimeSlots/TimeServiceTests.Exceptions.Modify.cs (hunk 2 above)
Comment (MirzayevDeveloper) [reply Tarteeb.Api.Tests.Unit/Services/Foundations/TimeSlots/TimeServiceTests.Exceptions.Modify.cs:149]: @bilolkobilov

File: Tarteeb.Api.Tests.Unit/Services/Foundations/TimeSlots/TimeServiceTests.Validation.Modify.cs (hunk 3 above)
Comment (MirzayevDeveloper) [reply Tarteeb.Api.Tests.Unit/Services/Foundations/TimeSlots/TimeServiceTests.Validation.Modify.cs:377]: @bilolkobilov

File: Tarteeb.Api.Tests.Unit/Services/Foundations/TimeSlots/TimeServiceTests.Exceptions.Modify.cs (hunk 2 above)
Comment (bilolkobilov) [reply Tarteeb.Api.Tests.Unit/Services/Foundations/TimeSlots/TimeServiceTests.Exceptions.Modify.cs]: Done.

File: Tarteeb.Api.Tests.Unit/Services/Foundations/TimeSlots/TimeServiceTests.Validation.Modify.cs (hunk 3 above)
Comment (bilolkobilov) [reply Tarteeb.Api.Tests.Unit/Services/Foundations/TimeSlots/TimeServiceTests.Validation.Modify.cs:378]: Done.

**Review by Javlonbek-dev — APPROVED** [review]

File: Tarteeb.Api.Tests.Unit/Services/Foundations/Times/TimeServiceTests.Validation.Modify.cs [hunk 4]
```diff
@@ -0,0 +1,382 @@
...
+                .BeEquivalentTo(expectedTimeValidationException);
+
+            this.loggingBrokerMock.Verify(broker =>
+                broker.LogError(It.Is(SameExceptionAs(
+                    expectedTimeValidationException))), Times.Once);
+
+            this.storageBrokerMock.Verify(broker =>
+                broker.UpdateTimeAsync(It.IsAny<Time>()), Times.Never);
+
+            this.storageBrokerMock.VerifyNoOtherCalls();
```
Comment (jasurbekyusuf) [inline Tarteeb.Api.Tests.Unit/Services/Foundations/Times/TimeServiceTests.Validation.Modify.cs:48]: Ketma ketlik noto'g'ri

File: Tarteeb.Api.Tests.Unit/Services/Foundations/Times/TimeServiceTests.Exceptions.Modify.cs [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+            this.storageBrokerMock.Verify(broker =>
+               broker.SelectTimeByIdAsync(someTime.Id), Times.Once);
+
+            this.loggingBrokerMock.Verify(broker =>
+                broker.LogError(It.Is(SameExceptionAs(
+                    expectedTimeServiceException))), Times.Once);
+
+            this.storageBrokerMock.VerifyNoOtherCalls();
+            this.dateTimeBrokerMock.VerifyNoOtherCalls();
+            this.loggingBrokerMock.VerifyNoOtherCalls();
```
Comment (jasurbekyusuf) [inline Tarteeb.Api.Tests.Unit/Services/Foundations/Times/TimeServiceTests.Exceptions.Modify.cs:215]: Buyerda ham

File: Tarteeb.Api.Tests.Unit/Services/Foundations/Times/TimeServiceTests.Validation.Modify.cs (hunk 4 above)
Comment (bilolkobilov) [reply Tarteeb.Api.Tests.Unit/Services/Foundations/Times/TimeServiceTests.Validation.Modify.cs]: Resolved

File: Tarteeb.Api.Tests.Unit/Services/Foundations/Times/TimeServiceTests.Exceptions.Modify.cs (hunk 5 above)
Comment (bilolkobilov) [reply Tarteeb.Api.Tests.Unit/Services/Foundations/Times/TimeServiceTests.Exceptions.Modify.cs:215]: Resolved

**Pull request merged**